tests/typerel/t4799_2.nim

```diff
@@ -18,4 +18,4 @@ proc testVehicle[T](x: varargs[Vehicle[T]]): string =
 var v = Vehicle[int](tire: 3)
 var c = Car[int](tire: 4)
 var b = Bike[int](tire: 2)
-echo testVehicle([b, c, v])
\ No newline at end of file
+echo testVehicle([v, c, b])
```

Comment:
Why was this necessary? The point of `commonType` is to compute the type to be `Vehicle[int]` regardless of the order in the array construction!

Reply:
see https://github.com/nim-lang/Nim/issues/7955#issuecomment-1251817322

Reply:
I don't agree with your comment.

Reply:
I'd say the issue case (`var y = [b, c] #failed`) is the worse case, neither previous nor current implementation fullfill your statement.

Reply:
to be clear, check `proc semArrayConstr` the implementation says order are matters. and I remenber `typeRel` only has `subType` to describe this, there's no things like `commonType` or `fatherType`.

Reply:
```
nimgrep commonType --ext:nim compiler/
compiler/sem.nim:126: template commonTypeBegin*(): PType = PType(kind: tyUntyped)
compiler/sem.nim:128: proc commonType*(c: PContext; x, y: PType): PType =
compiler/sem.nim:216: proc commonType*(c: PContext; x: PType, y: PNode): PType =
compiler/sem.nim:219:   commonType(c, x, y.typ)
compiler/semexprs.nim:660:       typ = commonType(c, typ, xx.typ)
compiler/semexprs.nim:2423:   var typ = commonTypeBegin
compiler/semexprs.nim:2441:           typ = commonType(c, typ, it[1].typ)
compiler/semexprs.nim:2457:           typ = commonType(c, typ, it[0].typ)
compiler/semstmts.nim:160:   var typ = commonTypeBegin
compiler/semstmts.nim:169:       typ = commonType(c, typ, it[1])
compiler/semstmts.nim:175:       typ = commonType(c, typ, it[0])
compiler/semstmts.nim:211:   var typ = commonTypeBegin
compiler/semstmts.nim:214:   typ = commonType(c, typ, n[0].typ)
compiler/semstmts.nim:274:       typ = commonType(c, typ, a[^1])
compiler/semstmts.nim:1075:   var typ = commonTypeBegin
compiler/semstmts.nim:1109:       typ = commonType(c, typ, x[last])
compiler/semstmts.nim:1117:       typ = commonType(c, typ, x[1])
compiler/semstmts.nim:1123:       typ = commonType(c, typ, x[0])
```

Reply:
hmm , I get it